### DISCUSSION TITLE
Input type password <i>visibility</i> clickable

### DISCUSSION THREAD

**Discussion by KicKerBNU** [beercss/beercss#382] [Ideas]
![image](https://github.com/user-attachments/assets/7ffa9f4d-fb20-4d0d-b3d4-b18296876782)

Right now we can add the <i>visibility</i> but we need to add some JS to change the state to make the password visible is it a good idea to make it automatically possible to click on the icon to turn the password visible ?

Workaround:
`<div class="field label border">`
`   <input :type="inputPasswordType">`
`   <label>Password</label>`
`   <a @click="changeInputType"><i>visibility</i></a>`
`</div>`

**Comment by laurentpayot** [comment]
@KicKerBNU 
>  is it a good idea to make it automatically possible to click on the icon to turn the password visible ?

You mean with Beercss’s included JS instead of custom JS? I don’t think it’s possible with pure CSS.

**Reply by KicKerBNU** [reply]
Yes, adding inside our JS file looks pretty common on the web, users are familiar with this behavior.

**Reply by laurentpayot** [reply]
It sounds reasonable to me, but as I’m only using the CSS part of Beercss, I will let @leonardorafael answer :wink:

**Reply by leonardorafael** [reply]
It's ok for me too

**Comment by coffeeandwork** [comment]
I also like the idea. There seems to be a way to accomplish showing/hiding password without using javascript but by using the [-webkit-text-security](https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-text-security) css property. This is clearly not the right way to go since the css property is not well supported, but I just wanted to bring attention to it. CSS support changes often, maybe in the future this (or another property like this) will be widely supported which will allow us to do this without javascript.

**Comment by ns8482e** [comment]
It seems it's not clickable because of z-index icon's z-index needs to be greater to react to pointer events

**Comment by leonardorafael** [comment]
Available in [v3.8.0](https://github.com/beercss/beercss/releases/tag/v3.8.0)

If you want a clickable icon, use `<a>` tag or `front` / `back` helper:

```html
<div class="field">
  <input type="password">
  <a>
    <i>visibility</i>
  </a>
</div>
```

```html
<div class="field">
  <input type="password">
  <i class="front">visibility</i>
</div>
```
